xsens-tf-bridge/src/xsens-tf-bridge.cpp

```diff
@@ -0,0 +1,262 @@
+    uint64_t nsec_part = (time % 1000000000UL);
+    uint64_t sec_part = (time / 1000000000UL);
+    TickTime ret;
+
+    if (sec_part > UINT_MAX)
+    {
+        yWarning() << "Timestamp exceeded the 64 bit representation, resetting it to 0";
+        sec_part = 0;
+    }
+
+    ret.sec  = sec_part;
+    ret.nsec = nsec_part;
+    return ret;
+}
+
+class MyModule : public RFModule, public yarp::os::TypedReaderCallback<xsens::XsensFrame>
+{
+    Port handlerPort; // a port to handle messages
+    Port client_port;
+    
+    Publisher<tf2_msgs_TFMessage> publisher_tf;
+    
+    yarp::os::BufferedPort<xsens::XsensFrame> xsensDataPort;
+    xsens::XsensDriverService xsensDriver;
+    
+    //variables declaration
+    int i, segmentsCount, trueSegmentCount;
+    
+    //tf_tfMessage tfMessage;
+    tf2_msgs_TFMessage tf;
+
+    std::string xsensServerName;
+    std::string driverServerName;
+    std::vector<std::string> segments;
+    
+public:
+    double getPeriod()
+    {
+        // module periodicity (seconds), called implicitly by the module.
+        return 0.1;
+    }
+    // This is our main function. Will be called periodically every getPeriod() seconds
+    bool updateModule()
+    {
+        cout << "waiting for input" << endl;
+        xsens::XsensFrame *xsensData = xsensDataPort.read();
+        if (xsensData!=NULL) {
+            cout << "got xsens data " << endl;
+            }
+
+        // Send the transforms
+        for (i=0;i<trueSegmentCount;i++){
+        tf.transforms[i].header.seq   = i;
+        tf.transforms[i].header.stamp = normalizeSecNSec(xsensData->time);
+        tf.transforms[i].transform.translation.x = xsensData->segmentsData[i].position.c1;
+        tf.transforms[i].transform.translation.y = xsensData->segmentsData[i].position.c2;
+        tf.transforms[i].transform.translation.z = xsensData->segmentsData[i].position.c3;
+        tf.transforms[i].transform.rotation.x = xsensData->segmentsData[i].orientation.c2;
+        tf.transforms[i].transform.rotation.y = xsensData->segmentsData[i].orientation.c3;
+        tf.transforms[i].transform.rotation.z = xsensData->segmentsData[i].orientation.c4;
+        tf.transforms[i].transform.rotation.w = xsensData->segmentsData[i].orientation.c1;
+        }
+        
+        for (i=(trueSegmentCount-1);i<segmentsCount;i++){
+        tf.transforms[i].header.seq   = i;
+        tf.transforms[i].header.stamp = normalizeSecNSec(xsensData->time);
+        tf.transforms[i].transform.translation.x = 0;
+        tf.transforms[i].transform.translation.y = 0;
+        tf.transforms[i].transform.translation.z = 0;
+        tf.transforms[i].transform.rotation.x = 0;
+        tf.transforms[i].transform.rotation.y = 0;
+        tf.transforms[i].transform.rotation.z = 0;
+        tf.transforms[i].transform.rotation.w = 1;
+        }
+
+        publisher_tf.write(tf);
+
+        i++;
+        for (i=0;i<segmentsCount;i++){
+        cout << tf.transforms[i].child_frame_id << endl;
```

Comment:
Please remove all the debug prints, or make those prints conditional on a `verbose` parameter.